server/services/v1/query_runner.go

```diff
@@ -1071,14 +1080,26 @@ func (runner *SubscribeQueryRunner) Run(ctx context.Context, tx transaction.Tx,
 		return nil, ctx, err
 	}
 
+	var partNums []int32
+	if runner.req.Options != nil && len(runner.req.Options.Partitions) > 0 {
+		partNums = runner.req.Options.Partitions
+	} else {
+		partNums = make([]int32, partitions)
+		for i := range partNums {
+			partNums[i] = int32(i)
+		}
+	}
+
 	type Part struct {
+		num       uint16
 		skipFirst bool
 		startTime time.Time
 	}
 
-	var parts [partitions]Part
+	parts := make([]Part, len(partNums))
 
-	for i := 0; i < len(parts); i++ {
+	for i := 0; i < len(partNums); i++ {
+		parts[i].num = uint16(partNums[i])
```

Comment:
Do we need to add a check here if part is valid?

Reply:
Yes, good catch.